packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/ProjectAssetService.java

```diff
@@ -52,20 +67,76 @@ public ProjectAsset findByProjectIdAndAssetIdAndAssetType(@NotNull final UUID pr
 		return projectAssetRepository.findByProjectIdAndAssetIdAndAssetType(projectId, assetId, type);
 	}
 
-	public ProjectAsset createProjectAsset(Project project, final AssetType type, final UUID assetId) {
+	private boolean populateProjectAssetFields(ProjectAsset projectAsset, AssetType assetType, UUID id)
+			throws IOException {
+		switch (assetType) {
+			case DATASET:
+				final Optional<Dataset> dataset = datasetService.getDataset(id);
+				if (dataset.isPresent()) {
+					projectAsset.setAssetName(dataset.get().getName());
+				}
```

Comment:
IntelliJ propose to do the following, is that better?
```suggestion
				dataset.ifPresent(dataset ->	projectAsset.setAssetName(dataset.get().getName()));
```

Reply:
It's just a different way to express it, the original code uses a condition, the suggestion uses a lambda.

Whats funny is the Intellij suggestion gives a warning in my VSCode for using a pre-existing variable name lol :)